cosmoz-default-tree.html

```diff
@@ -23,134 +35,146 @@
 
 	Cosmoz.DefaultTree.prototype = Object.create(Cosmoz.Tree.prototype);
 
-	Cosmoz.DefaultTree.prototype.getNodeByProperty = function (propertyName, propertyValue) {
+	/**
+	 * Returns the first found Node.
+	 */
+	Cosmoz.DefaultTree.prototype.getNodeByProperty = function (propertyName, propertyValue, exact, nodes) {
 		if (propertyName === undefined || propertyValue === undefined) {
 			return;
 		}
+		// Defaults
+		exact = exact !== undefined ? exact : true;
 
 		if (propertyName === 'pathLocator') {
-			return this._getNodeByPathLocator(propertyValue);
+			return this.getNodeByPathLocator(propertyValue);
 		}
 
-		return this._searchAllRoots(propertyName, propertyValue);
+		var results = this.searchNodes(propertyName, propertyValue, exact, nodes);
+		return results && results.length > 0 ? results[0] : null;
 	};
 
-	Cosmoz.DefaultTree.prototype.getPath = function (node) {
-		if (node === undefined) {
-			return;
-		}
-		return this._getPathByPathLocator(node.pathLocator);
+	/**
+	 * Loops over a node list (or roots if not defined) and returns the concatenated results of search().
+	 */
+	Cosmoz.DefaultTree.prototype.searchNodes = function (propertyName, propertyValue, exact, nodes) {
+		var results = [];
+
+		// Defaults
+		exact = exact !== undefined ? exact : true;
+		nodes = nodes ? nodes : this._roots;
+
+		nodes.forEach(function (node) {
+			results = results.concat(this.search(node, propertyName, propertyValue, exact));
+		}, this);
+			
+		return results;
 	};
 
-	Cosmoz.DefaultTree.prototype.getPathByProperty = function (propertyName, propertyValue) {
-		if (propertyName === undefined || propertyValue === undefined) {
-			return;
+	/**
+	 * Returns the Node of a given pathLocator.
+	 */
+	Cosmoz.DefaultTree.prototype.getNodeByPathLocator = function (pathLocator) {
+		if (!pathLocator) {
+			return this._treeData;
 		}
+		var	pathNodes = this.getPathNodes(pathLocator, this._treeData),
+			lastNodeOnPath = pathNodes.pop();
+		return lastNodeOnPath !== undefined ? lastNodeOnPath : null;
```

Comment:
`return pathNodes.pop() || null;`

But do we need to differentiate between `undefined` and `null` ?